src/modules/dashboard/components/ColonyHome/ColonyHome.tsx

```diff
@@ -143,32 +144,44 @@ const ColonyHome = ({ match, location }: Props) => {
     filteredDomainId || ROOT_DOMAIN_ID,
   ]);
 
-  const rootUserRoles = useTransformer(getUserRolesForDomain, [
-    data && data.colony,
+  const allUserRoles = useTransformer(getAllUserRoles, [
+    data?.colony,
     walletAddress,
-    ROOT_DOMAIN_ID,
   ]);
 
+  const handleUpgradeColony = useCallback(() => {
+    if (!data || !data.colony) {
+      return;
+    }
+    openUpgradeVersionDialog({
+      colony: data.colony,
+    });
+  }, [data, openUpgradeVersionDialog]);
+
   if (!colonyName || (reverseENSAddress as any) instanceof Error) {
     return <Redirect to={NOT_FOUND_ROUTE} />;
   }
 
   if (!data || !data.colonyAddress || !data.colony) {
     return <LoadingTemplate loadingText={MSG.loadingText} />;
   }
-
   const { colony } = data;
-  const canUpgradeColony = hasRoot(rootUserRoles);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+  const canUpgradeColony = hasRegisteredProfile && hasRoot(allUserRoles);
   /*
    * @NOTE As a future upgrade, we can have a mapping where we keep track of
    * past and current network versions so that we can control, more granularly,
    * which versions *must* be upgraded, and which can function as-is, even with
    * an older version
    */
   const mustUpgradeColony = canBeUpgraded(
-    data.colony,
+    colony,
     parseInt(networkVersion || '0', 10),
   );
+
+  const isSupportedColonyVersion = colony.version >= 5;
```

Comment:
Same here, please use the version constant from colonyJS

```suggestion
        const isSupportedColonyVersion = colony.version >= ColonyVersion.CeruleanLightweightSpaceship);
```